oasis-build/src/gen/imports.rs

```diff
@@ -294,6 +309,25 @@ fn get_rustc_version() -> String {
         .expect("Could not determine rustc version")
 }
 
+#[derive(Default)]
+struct HashableChecker {
+    contains_nonhashable: bool,
+}
+
+impl HashableChecker {
+    pub fn is_hash(&self) -> bool {
+        !self.contains_nonhashable
+    }
+}
+
+impl oasis_rpc::visitor::IdlVisitor for HashableChecker {
+    fn visit_type(&mut self, ty: &oasis_rpc::Type) {
+        use oasis_rpc::Type::{F32, F64};
+        self.contains_nonhashable |= *ty == F32 || *ty == F64;
```

Comment:
What's the meaning of non-hashable? And does it include only `F32` and `F64`?

Reply:
it's a type that can't `#[derive(Hash)]`. floats are impermissible because `NaN` is not `PartialEq`